mobile/lib/modules/trash/providers/trashed_asset.provider.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/home/ui/asset_grid/asset_grid_data_structure.dart';
+import 'package:immich_mobile/modules/trash/services/trash.service.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/models/exif_info.dart';
+import 'package:immich_mobile/shared/providers/asset.provider.dart';
+import 'package:immich_mobile/shared/providers/db.provider.dart';
+import 'package:immich_mobile/shared/providers/user.provider.dart';
+import 'package:isar/isar.dart';
+import 'package:logging/logging.dart';
+
+class TrashNotifier extends StateNotifier<bool> {
+  final Isar _db;
+  final Ref ref;
+  final TrashService _trashService;
+  final log = Logger('TrashNotifier');
+
+  TrashNotifier(
+    this._trashService,
+    this._db,
+    this.ref,
+  ) : super(false);
+
+  Future<void> emptyTrash() async {
+    try {
+      final user = ref.watch(currentUserProvider);
+      if (user == null) {
+        return;
+      }
+      await _trashService.updateTrash(deleteAll: true);
+
+      final assets = await _db.assets
+          .filter()
+          .ownerIdEqualTo(user.isarId)
+          .isTrashedEqualTo(true)
+          .findAll();
+
+      var dbIds = assets.where((a) => a.isRemote).map((e) => e.id).toList();
+
+      await _db.writeTxn(() async {
+        await _db.exifInfos.deleteAll(dbIds);
+        await _db.assets.deleteAll(dbIds);
+      });
+
+      // Refresh assets in background
+      Future.delayed(
+        const Duration(seconds: 4),
+        () async => await ref.read(assetProvider.notifier).getAllAsset(),
+      );
+    } catch (error, stack) {
+      log.severe("Cannot empty trash ${error.toString()}", error, stack);
+    }
+  }
+
+  Future<bool> restoreAssets(Iterable<Asset> assetList) async {
+    try {
+      final result = await _trashService.restoreAssets(assetList);
+
+      if (result) {
+        final remoteAssets = assetList.where((a) => a.isRemote).toList();
+
+        var updatedAssets = remoteAssets.map((e) {
+          e.isTrashed = false;
+          return e;
+        }).toList();
+
+        await _db.writeTxn(() async {
+          await _db.assets.putAll(updatedAssets);
```

Comment:
this will overwrite the assets in the DB, resulting in loss of connection between assets on device and remote assets. I suggest to only trigger a remote sync. This will take care of everything.